src/freenas/usr/local/bin/truenas-initrd.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+import contextlib
+import json
+import logging
+import os
+import subprocess
+import sys
+
+import libzfs
+import pyudev
+import sqlite3
+
+
+logger = logging.getLogger(__name__)
+
+
+def update_zfs_default(root):
+    with libzfs.ZFS() as zfs:
+        existing_pools = [p.name for p in zfs.pools]
+
+    for i in ['freenas-boot', 'boot-pool']:
+        if i in existing_pools:
+            boot_pool = i
+            break
+    else:
+        logger.error('Failed to locate valid boot pool. Pools located were: %r', ', '.join(existing_pools))
+        return False
+
+    with libzfs.ZFS() as zfs:
+        disks = [disk.replace("/dev/", "") for disk in zfs.get(boot_pool).disks]
+
+    mapping = {}
+    for dev in filter(
+        lambda d: not d.sys_name.startswith("sr") and d.get("DEVTYPE") in ("disk", "partition"),
+        pyudev.Context().list_devices(subsystem="block")
+    ):
+        if dev.get("DEVTYPE") == "disk":
+            mapping[dev.sys_name] = dev.get("ID_BUS")
+        elif dev.get("ID_PART_ENTRY_UUID"):
+            parent = dev.find_parent("block")
+            mapping[dev.sys_name] = parent.get("ID_BUS")
+            mapping[os.path.join("disk/by-partuuid", dev.get("ID_PART_ENTRY_UUID"))] = parent.get("ID_BUS")
+
+    has_usb = False
+    for dev in disks:
+        if mapping.get(dev) == "usb":
+            has_usb = True
+            break
+
+    zfs_config_path = os.path.join(root, "etc/default/zfs")
+    with open(zfs_config_path) as f:
+        original_config = f.read()
+        lines = original_config.rstrip().split("\n")
+
+    zfs_var_name = "ZFS_INITRD_POST_MODPROBE_SLEEP"
+    lines = [line for line in lines if not line.startswith(f"{zfs_var_name}=")]
+    if has_usb:
+        lines.append(f"{zfs_var_name}=15")
+
+    new_config = "\n".join(lines) + "\n"
+    if new_config != original_config:
+        with open(zfs_config_path, "w") as f:
+            f.write(new_config)
+        return True
+    return False
+
+
+def dict_factory(cursor, row):
```

Comment:
Let's put this to some middleware.utils module to avoid copy-pasting to every standalone console script we write